### DISCUSSION TITLE
Fitting SKTB Model on orthogonal basis

### DISCUSSION THREAD

**Discussion by favonein** [deepmodeling/DeePTB#313] [Q&A]
Hi,

I am having trouble fitting a SKTB model for just the graphene unit cell. I have tried starting from scratch and pushing the cutoff radius as well as starting from the empirical SK parameters, following the examples for hBN and Si, but the loss will stay around 0.6 at the lowest for either sp or spd* basis. I tried fitting an e3 model and a non-orthogonal model and both worked extremely well. I was wondering if there are any tips/tricks that you might have to help me. 

This is an example of the fit bands and the config I used.

<img width="785" height="582" alt="image" src="https://github.com/user-attachments/assets/f71122e2-55b8-49c4-918f-380f28eb280c" />

[train_config.json](https://github.com/user-attachments/files/25425914/train_config.json)

**Comment by favonein** [comment]
If possible, could you share the training config used for graphene/diamond model in the original DeePTB paper?

**Reply by AsymmetryChou** [reply]
Maybe you can try the baseline model inherent in DeePTB code to enhance the performance. 
For details, please refer to https://www.bohrium.com/notebooks/15254953382 and https://www.bohrium.com/notebooks/15262873382.

**Reply by AsymmetryChou** [reply]
Strating from scratch is not a good idea for DeePTB-SK model now.

**Reply by favonein** [reply]
I tried both from scratch and from the baseline empirical SK parameters, with no luck.

**Reply by AsymmetryChou** [reply]
It's abnormal since this is a simple task for DeePTB-SK. We have repeatly checked it. Something wrong may happen in your input files. If possible, you may upload all related files and data. And our group can check it completely.

**Reply by favonein** [reply]
Here is an example of the inputs I have been using. I have tested changing many of the settings including the info.json energy/band ranges, the onsite correction, training hyperparameter, etc.

[inputs_gr.tar.gz](https://github.com/user-attachments/files/25495331/inputs_gr.tar.gz)

**Reply by Lonya0** [reply]
it seems like your generated baseline model have no overlap? Can you discibe how did you get it? Usually it will have overlap parameters.

**Reply by favonein** [reply]
Hi,

I used overlap = false in the sk_in.json. I very specifically want to train the model on an orthogonal basis.

**Reply by Lonya0** [reply]
It will be very hard to training on orthogonal basis, you may firstly use a model with overlap, and with loss getting very low, then turn on push to make overlap decrease slowly to orthogonal. Otherwise the trainning will cost very long time(may be forever)

**Reply by favonein** [reply]
I tested this with good results for the unit cell. For training on MD snapshots and strongly perturbed structures, e.g. CNTs along with the flat graphene, would you recommend pushing the overlap factor to 0 before or after expanding the dataset from unit cell to the perturbed structures?
